meeting.js

```diff
@@ -0,0 +1,346 @@
+      console.log('Remote share started, video element should be visible');
+    }
+  });
+  meeting.on('media:stopped', (media) => {
+    console.log('media:stopped event triggered for type:', media.type);
+    if (media.type === 'local') {
+      document.getElementById('localVideo').srcObject = null;
+      document.getElementById('localVideo').style.backgroundImage = 'url("./images/image1.png")';
+    }
+    if (media.type === 'remoteVideo') {
+      document.getElementById('remoteVideo').srcObject = null;
+    }
+    if (media.type === 'remoteAudio') {
+      document.getElementById('remoteAudio').srcObject = null;
+    }
+    if (media.type === 'remoteShare') {
+      const remoteShareElement = document.getElementById('remoteShare');
+      remoteShareElement.srcObject = null;
+      console.log('Remote share stopped, video element should be hidden');
+    }
+  });
+}
+/*document.getElementById('remoteoff').addEventListener('click',function(){
+  document.getElementById('remoteShare').style.display="none";
+  document.getElementById('remoteoff').style.display="none";
+  const remoteVideoElement = document.getElementById('remoteVideo');
+      remoteVideoElement.style.width = '846px'; 
+      remoteVideoElement.style.height = '486px'; 
+
+})
+document.getElementById('remotewidth').addEventListener('click',function(){
+  const remoteVideoElement = document.getElementById('remoteVideo');
+  remoteVideoElement.style.width = '320px';  // Set width to 320px
+  remoteVideoElement.style.height = '240px'; // Set height to 240px
+  document.getElementById('remotewidth').style.display='none';
+})*/
+
+// Clean up media listeners
+function cleanUpMediaListeners(meeting) {
+  meeting.off('media:ready');
+  meeting.off('media:stopped');
+}
+
+// Including Leave Meeting logic
+document.getElementById('leaveMeetingButton').addEventListener('click', () => {
+  if (currentMeeting) {
+    const confirmLeave = window.confirm("Are you sure you want to leave the meeting?");
+    if (confirmLeave) {
+      currentMeeting.leave().then(() => {
+        document.getElementById('joinMeetingModal').style.display="none";
+        console.log('Left the meeting');
+        cleanUpMediaListeners(currentMeeting);
+        document.getElementById('Video').innerHTML='<img src="./images/stop.png" style="width: 25px; ;height:25px">Start'
+      }).catch((error) => {
+        console.error('Failed to leave the meeting', error);
+      });
+    } else {
+      console.log('User chose not to leave the meeting');
+      document.getElementById('remoteShare').style.display="none";
+    }
+  }
+});
+
+
+// Adding Event Listeners to the JOIN, START/STOP, UNMUTE/MUTE buttons
+document.addEventListener("DOMContentLoaded", () => {
+  document.getElementById("joinMeetingButton").addEventListener("click", joinMeeting);
+  document.getElementById("Video").addEventListener("click", toggleVideo);
+  document.getElementById("Microphone").addEventListener("click", toggleMicrophone);

```

Comment:
While you have added these event listeners on `DOMContentLoaded` why is `leaveMeetingButton` gets an event callback added differently in line number 94?

Reply:
Kesava these were loaded when the meeting occurred and the leave meeting button occurs to showcase the current so I made it when the document is loaded and join meeting comes under div one so I included it we can make it directly in the function itself for better and differentiate with the parameters are loaded before and after meeting.